so-widgets-bundle.php

```diff
@@ -98,37 +102,56 @@ function plugin_version_check(){
 
 		$active_version = get_option( 'siteorigin_widget_bundle_version' );
 
-		if( empty($active_version) || version_compare( $active_version, SOW_BUNDLE_VERSION, '<' ) ) {
-			// If this is a new version, then clear the cache.
-			update_option( 'siteorigin_widget_bundle_version', SOW_BUNDLE_VERSION );
+		$is_new = empty($active_version) || version_compare( $active_version, SOW_BUNDLE_VERSION, '<' );
+		$is_new = apply_filters( 'siteorigin_widgets_is_new_version', $is_new );
 
-			// Remove all cached CSS for SiteOrigin Widgets
-			if( function_exists('WP_Filesystem') && WP_Filesystem() ) {
-				global $wp_filesystem;
-				$upload_dir = wp_upload_dir();
-
-				// Remove any old widget cache files, if they exist.
-				$list = $wp_filesystem->dirlist( $upload_dir['basedir'] . '/siteorigin-widgets/' );
-				if( !empty($list) ) {
-					foreach($list as $file) {
-						// Delete the file
-						$wp_filesystem->delete( $upload_dir['basedir'] . '/siteorigin-widgets/' . $file['name'] );
-					}
-				}
-			}
+		if( $is_new ) {
 
-			// An action to let widgets handle the updates.
+			update_option( 'siteorigin_widget_bundle_version', SOW_BUNDLE_VERSION );
+			// If this is a new version, then trigger an action to let widgets handle the updates.
 			do_action( 'siteorigin_widgets_version_update', SOW_BUNDLE_VERSION, $active_version );
```

Comment:
Maybe a good compromise would be to have something like this when we detect a version number change.

`if( !has_action( 'upgrader_process_complete' ) && !has_action( 'activated_plugin' ) ) { /* clear all the caches */ }`

Would rather err on the side of caution for something like this incase there's an edge case where those 2 actions aren't called on a plugin update. Some examples of possible edge cases

* A user just deleting the `so-widgets-bundle` folder, then reuploading it from a zip file through FTP. They just leave the plugin active so a `activated_plugin` isn't triggered and they don't go through the updater so `upgrader_process_complete` isn't triggered.
* Possible that something like WP-CLI doesn't trigger these actions when doing a command line plugin update.
* There might be some 3rd party ManageWP-style WordPress management plugins/services that don't call these actions when a plugin is updated. I assume ManageWP is, but there might be other similar services that aren't.

We need to make sure the core widgets are handled properly and offer as much cache clearing as possible to 3rd party addons.